azurelinuxagent/common/protocol/restapi.py

```diff
@@ -144,26 +144,61 @@ def __init__(self):
         self.vmAgentManifests = DataContractList(VMAgentManifest)
 
 
+class Dependency(object):
+    """
+    Represents an extension upon which another extension is dependent
+    handler - the ExtHandler of the dependency
+    exts - the Extension(s) of the dependency
+    timeout - the timeout for the extension to get to the Ready state
+              timeout is expressed in minutes.
+    """
+
+    def __init__(self, handler=None, exts=None, timeout=None):
+        self.handler = handler
+        self.exts = exts
+        self.timeout = timeout
+
+
 class Extension(DataContract):
     def __init__(self,
                  name=None,
                  sequenceNumber=None,
                  publicSettings=None,
                  protectedSettings=None,
-                 certificateThumbprint=None):
+                 certificateThumbprint=None,
+                 dependencyLevel=0,
+                 dependencies=None):
```

Comment:
Why don't you just default to []?  You remove the need for the if statement below, and null arrays are evil. 😝

Reply:
It is a Python gotcha. If I used `[]` instead of `None`, and simplified the logic below, then _every_ new instance of `Extension` gets a pointer to the same default list, so an append to the `dependencies` of one `Extension` is an append to the `dependencies` of every `Extension` that was instantiated with the default parameter.

I don't think I run into that gotcha here given the way `Extension`s are currently being built, but I'd rather proactively avoid WTF debugging sessions where I can.